pipeline_runner/runner.py

```diff
@@ -252,8 +253,17 @@ def _get_image(self) -> Image:
 
         return Image(name=DEFAULT_IMAGE)
 
-    def _create_network(self) -> Network:
+    def _get_network(self) -> Network:
         name = f"{self._ctx.pipeline_ctx.project_metadata.slug}-network"
+        try:
+            bridge_network = self._docker_client.networks.get(name, driver="bridge")
+        except HTTPError:
+            print("Network {} doesn't exist yet...".format(name))
+        else:
+            bridge_network = self._create_network(name)
+        return bridge_network
+
+    def _create_network(self, name) -> Network:
```

Comment:
You got the right idea here, but some small modifications are needed. Basically:
```suggestion
    def _get_network(self) -> Network:
        name = f"{self._ctx.pipeline_ctx.project_metadata.slug}-network"
        try:
            bridge_network = self._docker_client.networks.get(name)
        except APIError as e:
            if e.status_code != HTTPStatus.NOT_FOUND:
                raise

            logger.debug("Creating network %s.", name)
            bridge_network = self._create_network(name)
        else:
            logger.debug("Network %s already exists.", name)
        return bridge_network

    def _create_network(self, name: str) -> Network:
```
The rationale behind the changes:

* The main change is the logic with try / except / else. The `else` executes only if there is no exception. In this case, it means we'd try to create the network if it already exists. The create network should be called in the except block. 
* The exception is now more narrow. If we get an APIError from the lib _and_ it's a NOT_FOUND, that means the network doesn't exist and we create it. Otherwise, there might be a legit error, in which case we raise.
* The print has been changed to a log entry instead.
* networks.get() takes only 1 param, the name
* This project enforces typing, so the `name` argument to `_create_network` must be typed.

I realize I've basically given you the code, but I'll let you update the PR so the credit is yours!